analyzer/src/funTest/kotlin/curation/ClearlyDefinedPackageCurationProviderFunTest.kt

```diff
@@ -20,10 +20,9 @@
 package org.ossreviewtoolkit.analyzer.curation
 
 import io.kotest.core.spec.style.WordSpec
-import io.kotest.matchers.maps.beEmpty
-import io.kotest.matchers.maps.haveSize
+import io.kotest.matchers.collections.beEmpty
```

Comment:
Commit message proposal as discussed:

```
refactor(PackageCurationProvider)!: Make `getCurationsFor()` return a list

Previously, `getCurationsFor()` return a map keyed by the `Identifier` of
the `Package` that was used to request the curation. For curations that
match multiple ids, e.g. because the use of version ranges, this resulted
in the same curation being returned multiple times for different ids.

To avoid these duplicated, and because all callers of `getCurationsFor()`
anyway call `flatten()` on the maps's values only, drop the keys by simply
returning a list instead.

Note that the `PackageCuration.isApplicable()` needs to happen
independently of that before writing curation to the ORT result, as there
is no guarantee that all curation provider implementations only return
applicable curations in the ORT sense.
```